File: backend/models/company.js
const mongoose = require('mongoose')

const Schema = mongoose.Schema
const ObjectId = Schema.ObjectId

const CompanySchema = new Schema({
	owner: { type: ObjectId, ref: 'User' },
	employees: [{ type: ObjectId, ref: 'User' }],
	name: String,
	location: {
		lat: Number,
		lon: Number
	},
	logo: String,
	description: String,
	goods: [{ type: ObjectId, ref: 'Good' }]
})

CompanySchema.methods = {

}

mongoose.model('Company', CompanySchema)

File: flutter/hackathome_frontend/lib/models/company.dart
import 'loc.dart';

class Company {
  String name;
  Loc location;
  String logo;
  String description;
  String id;

  Company.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        description = json['description'],
        logo = json['logo'],
        location = Loc.fromJson(json['location']),
        name = json['name'];

  Map<String, dynamic> toJson() => {
      "id":id,
      "name":name,
      "description":description,
      "logo":logo,
      "location":location.toJson()
  };
}
